### DISCUSSION TITLE
v0.0.2-beta

### DISCUSSION THREAD

**Discussion by jay-dee7** [containerish/OpenRegistry#55] [Announcements]
# Release Day 🔥 🔥 🔥 
We have our first public beta release ready, Here are the changes from the last release:

## What's Changed
* Token auth by @jay-dee7 in https://github.com/containerish/OpenRegistry/pull/48
* fix: bug fix for token auth by @jay-dee7 in https://github.com/containerish/OpenRegistry/pull/49
* add: enhanced sign-in experience by @guacamole in https://github.com/containerish/OpenRegistry/pull/51
* add: added better error responses by @guacamole in https://github.com/containerish/OpenRegistry/pull/52
* add: added api support for fetching registry images by @guacamole in https://github.com/containerish/OpenRegistry/pull/53
* Update README.md by @guacamole in https://github.com/containerish/OpenRegistry/pull/54

**Full Changelog**: https://github.com/containerish/OpenRegistry/compare/v0.0.1-beta...v0.0.2-beta

**Comment by redsolver** [comment]
Hi, I had some issues when logging in with the provided credentials. The generated password included `&gt` which was displayed as `>` in my email client. Obviously the credentials did not work and I had to look at the email plaintext source to find the correct password. A solution would be to not use special characters in the generated password (it is long enough anyways) or correctly escape the special characters in the HTML email.

**Comment by redsolver** [comment]
Another small issue I encountered is that when logging in using email in the Docker CLI it works, but pushing an image fails. And because my email starts with `info@red...`, my username was set to `info`, which was a bit confusing :)

**Comment by redsolver** [comment]
I just pushed an image to OpenRegistry and successfully pulled and deployed it on my home server! It worked pretty well, but the download speeds were a bit slow. I guess you are downloading the blob from siasky.net server-side and then serving it to clients. Maybe it would make sense to instead redirect the blob download to the siasky.net skylink directly to save bandwidth and improve performance.

**Comment by jay-dee7** [comment]
Hey @redsolver all these points make sense. Thanks a ton for sharing your feedback. 
We're going to make fixes for all the issues mentioned in the comments.

We actually didn't check it out but I remember now that only certain special characters are password-safe, we'll be moving on to link based signup process so that users can set their own passwords.

For silly usernames, we actually didn't store any user specific information other than their emails which of course isn't the best way. We'll be switching to random docker container like username for beta users and provide an api and UI/UX flow for users to change it anything they like

And finally, we already have plans for shifting the bandwidth to client side to save a ton of performance and our network bandwidth. Next release probably will have that

**Comment by jay-dee7** [comment]
@guacamole can we create action items for both of the above points? I believe we already have one for offloading the bw to client side

**Reply by guacamole** [reply]
Sure @jay-dee7 , I'm on it!

**Reply by guacamole** [reply]
Hi @jay-dee7 , The bandwidth thing is done, we're now offloading it to client side.  We have created issues for rest two!

**Reply by jay-dee7** [reply]
That was a pretty quick turn around

**Comment by bengrissinger** [comment]
The on-boarding was easy, Once I was able to read that the user name that has privilege to create a registry was the username handed to me in the email I received.  was able to easily push / pull images as expected.   nice work!
